### PR TITLE
add vault history to subgraph

### PR DESCRIPTION
**Pull request by EstherSeyi**
# Task: Add vault history to subgraph

## Description

This PR adds vault history the subgraph

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

You can test with https://thegraph.com/hosted-service/subgraph/estherseyi/squeeth-subgraph

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Review by KMKoushik — CHANGES_REQUESTED** [review]

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -40,4 +40,18 @@ type DayStatSnapshot @entity {
   id: ID!
   timestamp: BigInt!
   totalCollateralAmount: BigInt!
-}
\ No newline at end of file
+}
+
+enum Type {
+  WITHDRAWAL
```
[the remark this answers is not in the document]
Comment (EstherSeyi) [reply packages/subgraph/schema.graphql:46]: okay, will update

File: packages/subgraph/schema.graphql [hunk 2]
```diff
@@ -40,4 +40,18 @@ type DayStatSnapshot @entity {
...
+  DEPOSIT
+}
+
+type VaultTransaction @entity {
+  id: ID!
+  timestamp: BigInt!
+  totalCollateralAmount: BigInt!
+  type: Type!
+  vaultId: BigInt!
+  amount: BigInt!
```
[the remark this answers is not in the document]
Comment (EstherSeyi) [reply packages/subgraph/schema.graphql:56]: oh okay. So I should remove the amount field and instead have two fields ethCollatAmount and oSqthAmount right? Also is the shortAmount in the vault the oSqthAmount here?

[the remark this answers is not in the document]
Comment (KMKoushik) [reply packages/subgraph/schema.graphql:56]: In the mint action amount oSQTH minted is the sqthAmount in burn action oSQTH burnt is sqthAmount. That will actually have a meaningful use than vault shortAmount

**Review by KMKoushik — APPROVED** [review]

File: packages/subgraph/src/controller.ts [hunk 3]
```diff
@@ -70,213 +77,314 @@ import { loadOrCreateAccount, BIGINT_ONE, BIGINT_ZERO } from "./util"
...
+  transactionHash: string,
+  debtAmount: BigInt
+): VaultHistory {
+  const vaultHistory = new VaultHistory(transactionHash);
+  vaultHistory.totalEthCollateralAmount = vault.collateralAmount;
+  vaultHistory.action = action;
+  vaultHistory.vaultId = vaultId;
+  vaultHistory.timestamp = timestamp;
+  if (action === "MINT_SHORT" || action === "BURN_SHORT") {
+    vaultHistory.oSqthAmount = amount;
```
Comment (yichiehliu) [inline packages/subgraph/src/controller.ts:381]: Don't we need to add vaultHistory.ethCollateralAmount = amount for MINT_SHORT & BURN_SHORT tx types?

Comment (EstherSeyi) [reply packages/subgraph/src/controller.ts:381]: The amount in the event params for MINT_SHORT & BURN_SHORT is the osqth minted & burnt respectively. No ethCollateral amount there.

Comment (KMKoushik) [reply packages/subgraph/src/controller.ts:381]: Yep that's right

**Review by yichiehliu — APPROVED** [review]

**Pull request merged**